bonus_material/GPT_neo_IMDB_multinode/run_no_torchrun.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#SBATCH --account=project_465001063
+#SBATCH --partition=standard-g
+#SBATCH --nodes=2
+#SBATCH --gpus-per-node=8
+#SBATCH --ntasks-per-node=8  # we want one process per GPU
+#SBATCH --cpus-per-task=7
+#SBATCH --mem-per-gpu=60G
+#SBATCH --time=0:15:00
+
+
+# Set up the software environment
+# NOTE: these modules will be available from the LUMI system stack after July 2024 and the "module use" line will no longer work
+module purge
+module use /project/project_465001063/modules
+module load singularity-userfilesystems singularity-CPEbits
+
+CONTAINER=/scratch/project_465001063/containers/pytorch_transformers.sif
+
+# Some environment variables to set up cache directories
+SCRATCH="/scratch/${SLURM_JOB_ACCOUNT}"
+FLASH="/flash/${SLURM_JOB_ACCOUNT}"
+export TORCH_HOME=$SCRATCH/torch-cache
+export HF_HOME=$FLASH/hf-cache
+mkdir -p $TORCH_HOME $HF_HOME
+
+# Disable internal parallelism of huggingface's tokenizer since we
+# want to retain direct control of parallelism options.
+export TOKENIZERS_PARALLELISM=false
+
+# Path to where the trained model and logging data will go
+export OUTPUT_DIR=$SCRATCH/$USER/data/
+export LOGGING_DIR=$SCRATCH/$USER/runs/
+export MODEL_NAME=gpt-imdb-model-${SLURM_JOBID}
+
+set -xv # print the command so that we can verify setting arguments correctly from the logs
+
+# Set up variables to control distributed PyTorch training
+export MASTER_ADDR=$(scontrol show hostname ${SLURM_NODELIST} | head -n 1)
+export MASTER_PORT=25900
+export WORLD_SIZE=$SLURM_NPROCS
+export LOCAL_WORLD_SIZE=$WORLD_SIZE
+
```

Comment:
I believe you need to set `export NCCL_SOCKET_IFNAME=hsn` here as well when using the `aws-ofi-rccl` plugin.

Reply:
So the container comes with the plugin but doesn't set the variable already?

Reply:
I don't think so. It seems that @sfantao only sets it [when testing the containers](https://github.com/search?q=repo%3Asfantao%2Flumi-containers%20NCCL_SOCKET_IFNAME&type=code). It's an interesting idea though, so I will add it to the agenda for the next LUMI container group meeting to discuss if this should be set in containers that have the aws-ofi-rccl plugin installed. The [CXI_* variables are already set that way](https://github.com/sfantao/lumi-containers/blob/aeeb8993208ac05ed2294eaf44b31cf0f7be88a8/common/Dockerfile.aws-ofi-rccl#L25).

Reply:
But if you can run this without setting `NCCL_SOCKET_IFNAME`, it makes me wonder when that environment variable is really needed...

Reply:
Iirc, not setting this will still allow a multi-node run to work but it might use tcp instead of slingshot, so poorer performance results. I'll set it but I think it should be set in the container because it's as much part of proper network setup as the `aws-ofi-rccl` plugin.

Reply:
done